### PR TITLE
Multi-platform event API

### PR DESCRIPTION
**Pull request by Lennart99**
TODO:
- ~priorities~
- possibly ignoreCancelled

### CODE REVIEW

**Comment by Lennart99** [comment]
Created a wiki here: https://github.com/KevinDaGame/VoxelSniper-Reimagined/wiki/Event-API

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/events/player/PlayerBrushChangedEvent.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class PlayerBrushChangedEvent extends PlayerEvent {
+
+    private static final HandlerList<PlayerBrushChangedEvent> handlers = new HandlerList<>();
+    private final IBrush originalBrush;
+    private final IBrush newBrush;
+    private final String toolId;
+
+    public PlayerBrushChangedEvent(IPlayer p, String toolId, IBrush originalBrush, IBrush newBrush) {
+        super(p);
+        this.originalBrush = originalBrush;
```
[the remark this answers is not in the document]
Comment (Lennart99) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/events/player/PlayerBrushChangedEvent.java]: https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/81/commits/7c5186260aed6c467242956bccc03ac01253b1e5

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/events/player/PlayerBrushSizeChangedEvent.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+public class PlayerBrushSizeChangedEvent extends PlayerEvent {
+
+    private static final HandlerList<PlayerBrushSizeChangedEvent> handlers = new HandlerList<>();
+    private final int originalSize;
+    private final int newSize;
+    private final String toolId;
+
+    public PlayerBrushSizeChangedEvent(IPlayer p, String toolId, int originalSize, int newSize) {
+        super(p);
+        this.originalSize = originalSize;
```
[the remark this answers is not in the document]
Comment (Lennart99) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/events/player/PlayerBrushSizeChangedEvent.java]: https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/81/commits/7c5186260aed6c467242956bccc03ac01253b1e5

File: VoxelSniperCore/src/main/java/com/github/kevindagame/snipe/Sniper.java [hunk 3]
```diff
@@ -128,26 +129,21 @@ private boolean handleRightClick(IBlock clickedBlock, SnipeTool sniperTool, Snip
         }
 
         var success = sniperTool.getCurrentBrush().perform(snipeAction, snipeData, targetBlock, lastBlock);
-        VoxelSniper.voxelsniper.getEventManager().callSniperSnipeEvent(this, sniperTool.getCurrentBrush(), success);
+        PlayerSnipeEvent event = new PlayerSnipeEvent(this.getPlayer(), sniperTool.getCurrentBrush(), success);
```
[the remark this answers is not in the document]
Comment (Lennart99) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/snipe/Sniper.java]: https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/81/commits/ba55f4f12c01c85b82d87b4cecf3fa4969027c6f

File: VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/bstats/BrushUsageCounter.java [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
 
-public class BrushUsageCounter implements Listener {
+public class BrushUsageCounter {
 //    private static final Map<String, Integer> perBrushCounter = new HashMap<>();
     private static int counter = 0;
 
-    @EventHandler
-    private void onBrushUse(SniperSnipeEvent event) {
+    public void registerListeners() {
+        PlayerSnipeEvent.getHandlerList().registerListener(this::onBrushUse);
```
[the remark this answers is not in the document]
Comment (Lennart99) [reply VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/bstats/BrushUsageCounter.java]: Made quite some changes, I am not 100% sold on it yet. https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/81/commits/fd34a7fb96f5080339ae1dc49d666a4bc2dc03e3

**Review by KevinDaGame — APPROVED** [review]

**Pull request merged**